### PR TITLE
fix: displayed an alert if no public email provided in Github account; ver…

### LINKED ISSUE #280: Troubleshoot why first account verification email doesn't work for oAuth registered accounts
To recreate: 

- Create an account on [the Mobility Database](https://beta.mobilitydatabase.org/) with either Github or Google. 
- Submit registration form 
- Check email for verification email
- Email is not sent 
- Resend email
- You should receive email

### PR DESCRIPTION
**Pull request by qcdyx**
**Summary:**
Closes #280 
display an alert if user doesn't have public email in GitHub account
image
If user has public email in GitHub account, verify email and provide feedback of email verification
image

**Testing tips:**
Case 1: No Public email in GitHub account

- Go to https://github.com/settings/profile
- Public Email section, choose "Select a verified email to display", click update profile button
- goto MobilityDatabase API,  click signup with Github, you should see an alert "No public email provided in Github account. Please use a different registration method." as in screenshot 1.

image

Case 2: user has a public email in GitHub account

- go to https://github.com/settings/profile
- Public Email section, choose an email as public email, click update profile button
- goto MobilityDatabase API,  click signup with Github, you should see an message about the status of email verification, either Success or Error, as in screenshot 2.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: web-app/src/app/screens/SignUp.tsx [hunk 1]
```diff
@@ -105,7 +110,16 @@ export default function SignUp(): React.ReactElement {
...
     signInWithPopup(auth, provider)
       .then((userCredential: UserCredential) => {
-        dispatch(loginWithProvider({ oauthProvider, userCredential }));
+        if (!userCredential.user.emailVerified) {
+          dispatch(verifyEmail());
+        }
+        if (userCredential.user.email == null) {
+          alert(
+            'No public email provided in Github account. Please use a different registration method.',
+          );
```
Comment (davidgamez) [inline web-app/src/app/screens/SignUp.tsx:119]: [suggestion]: Use a Snackbar component with no _autoHideDuration_ set instead of JS alert.

**Comment by qcdyx** [comment]
@emmambd @davidgamez 
image

**Comment by davidgamez** [comment]
@qcdyx can you test the same flow starting from the `Sign-in` screen instead of the `Sign-up`

**Comment by emmambd** [comment]
@davidgamez #300 will resolve this one

**Comment by davidgamez** [comment]
> @davidgamez #300 will resolve this one
Yes! thanks!

**Review by davidgamez — APPROVED** [review]

**Pull request merged**